source/SoundCore.cpp

```diff
@@ -72,6 +72,102 @@ VOICE::~VOICE(void)
 	}
 }
 
+#ifdef _MSC_VER
+
+class DSSoundBuffer : public SoundBufferBase
```

Comment:
maybe this class should go to the Windows folder with (most of the) other Windows only code.